Tactics/Reflect/FetchAndDecode.lean

```diff
@@ -47,17 +56,67 @@ def reduceFetchInst? (addr : Expr) (s : Expr) :
         mkApp (.const ``BitVec []) (toExpr 32))
       (toExpr (some rawInst))
   ]
-  trace[Sym.reduceFetchInst] "{Lean.checkEmoji} found a proof:\n\t{proof}"
+  trace[Sym.reduceFetchDecode] "{Lean.checkEmoji} found a proof:\n\t{proof}"
   return ⟨rawInst, proof⟩
 
+
+
+/-! ## `reduceDecodeInst?` -/
+
+/-- `canocalizeBitVec e` recursively walks over expression `e` to convert any
+occerences of:
+  `BitVec.ofFin w (Fin.mk x _)`
+to the canonical form:
+  `BitVec.ofNat w x` (i.e., `x#w`)
+ -/
+-- TODO: should this canonicalize to `BitVec.ofNatLt` instead,
+--       as the current transformation loses information?
+partial def canonicalizeBitVec (e : Expr) : MetaM Expr := do
+  match_expr e with
+    | BitVec.ofFin w i =>
+        let_expr Fin.mk _ x _h := i | fallback
+        let w ←
+          if w.hasFVar || w.hasMVar then
+            pure w
+          else
+            withTransparency .all <| reduce w
```

Comment:
```suggestion
            withTransparency .all <| reduce w -- NOTE: potentially expensive reduction
```

I'd like to flag calls that trigger reduction to make it easy to debug performance problems.